src/core/dbus-service.c

```diff
@@ -135,12 +135,12 @@ static int bus_service_method_mount(sd_bus_message *message, void *userdata, sd_
 
         assert(message);
 
-        if (!MANAGER_IS_SYSTEM(u->manager))
-                return sd_bus_error_set(error, SD_BUS_ERROR_NOT_SUPPORTED, "Adding bind mounts at runtime is only supported by system manager");
-
-        r = unit_can_live_mount(u, error);
-        if (r < 0)
-                return r;
+        if (!unit_can_live_mount(u, error))
+                return log_unit_debug_errno(
+                                u,
+                                error ? sd_bus_error_get_errno(error) : SYNTHETIC_ERRNO(EINVAL),
+                                "Unit does not support live mounting: %s",
+                                bus_error_message(error, EINVAL));
```

Comment:
Otherwise, if `error` is NULL, this always logs with EINVAL, even the checker returns false with other reasons.

Reply:
Or, always pass a sd_bus_error object to `unit_can_live_mount()` even if `error` is NULL.

Reply:
Why would that matter? The only thing that matters is that if there is something worth logging, it happens. There's no case where there isn't, so it makes no difference

Reply:
I've changed it to EOPNOTSUPP which is accurate. This should be more than enough.

Reply:
I am concerning about that the error code logged here depends on whether `error` is NULL or not.
If you want to make `_can_live_mount()` return bool (again, I agree with that if the name is kept), then please always pass non-NULL `sd_bus_error` object here. Then, the logged error code is always consistent.
```c
_cleanup_(sd_bus_error_free) sd_bus_error error2 = SD_BUS_ERROR_NULL;
sd_bus_error *e = error ?: &error2;
if (!unit_can_live_mount(u, e))
        return log_unit_debug_errno(...);
```

Reply:
> I've changed it to EOPNOTSUPP which is more accurate. This should be enough for a hypothetical fail-safe fallback?

Though the change is not necessary when the above suggestion is applied, the change should be moved to the first commit.

Reply:
> I am concerning about that the error code logged here depends on whether error is NULL or not.

That doesn't happen anywhere, so it doesn't matter. And if it were to happen, "operation not supported" is a correct message to return as a safety fallback. There is no need for anything more complex than that.

Reply:
I do not think my suggestion is complex. Only two lines needs to be inserted, and then you can drop conditions in `log_unit_debug_errno()`, that's much simpler I think. I have not checked when `error` is NULL. But, at least, the function does not have `assert(error)`, so potentially it is NULL. I cannot catch why you want to stick to the current form.

I do not accept the current form. Though, that should be mostly cosmetic, hopefully. So, if other reviewers accept the current form, please go ahead. @YHNdnzj @poettering @DaanDeMeyer Please consider that point.

Reply:
> I have not checked when error is NULL.

I'm not aware of any case where it would. The fallbacks have to be provided because that's how these things work, and EOPNOTSUPP is just fine for that purpose, especially for a debug level message. Can we please get this merged so the debug level log spam goes away, given nothing in this discussion has any real functional impact and it's purely cosmetic? Thanks